src/app/core/services/unit.service.ts

```diff
@@ -5,43 +5,40 @@ import { Injectable } from '@angular/core';
 
 declare const window: any;
 
+type TempUnit = Omit<UnitDetails, 'stats' | 'ability' | 'target' | 'pattern' | 'special'>;
```

Comment:
lets give this type a proper name

Reply:
i dont have a better name for it 😄 
its rly a temporary object that represents a unit that does not have some properties yet and was the best TS-way i could do it